magenta-lib/src/main/scala/magenta/deployment_type/GCS.scala

```diff
@@ -0,0 +1,158 @@
+       the `prefixStage`, `prefixPackage` and `prefixStack` keys - none of those prefixes will be applied, as you have
+       full control over the path with the resource lookup.
+    """.stripMargin,
+    optional = true
+  )
+
+  //required configuration, you cannot upload without setting these
+  val bucket = Param[String]("bucket", "GCS bucket to upload package files to (see also `bucketResource`)", optional = true)
+  val bucketByStage: Param[Map[String, String]] = Param[Map[String, String]](
+    name = "bucketByStage",
+    documentation =
+      """
+        |A dict of stages to buckets in the package. When the current stage is found in here the bucket
+        |will be used from this dict. If it's not found here then it will fall back to `bucket` if it exists.
+        |```
+        |bucketByStage:
+        |  PROD: prod-bucket
+        |  CODE: code-bucket
+        |```
+        |""".stripMargin,
+    optional = true
+  )
+
+  val publicReadAcl = Param[Boolean]("publicReadAcl",
+    "Whether the uploaded artifacts should be given the PublicRead Canned ACL. (Default is true!)"
+  ).default(true)
+
+  val cacheControl = Param[List[PatternValue]]("cacheControl",
+    """
+      |Set the cache control header for the uploaded files. This can take two forms, but in either case the format of
+      |the cache control value itself must be a valid HTTP `Cache-Control` value such as `public, max-age=315360000`.
+      |
+      |In the first form the cacheControl parameter is set to a JSON string and the value will apply to all files
+      |uploaded to the GCS bucket.
+      |
+      |In the second form the cacheControl parameter is set to an array of JSON objects, each of which have `pattern`
+      |and `value` keys. `pattern` is a Java regular expression whilst `value` must be a valid `Cache-Control` header.
+      |For each file being uploaded, the array of regular expressions is evaluated against the file path and name. The
+      |first regular expression that matches the file is used to determine the cache control value. If there is no match
+      |then no cache control will be set.
+      |
+      |In the example below, if the file path matches either of the first two regular expressions then the cache control
+      |header will be set to ten years (i.e. never expire). If neither of the first two match then the catch all regular
+      |expression of the last object will match, setting the cache control header to one hour.
+      |
+      |    "cacheControl": [
+      |      {
+      |        "pattern": "^js/lib/",
+      |        "value": "max-age=315360000"
+      |      },
+      |      {
+      |        "pattern": "^\d*\.\d*\.\d*/",
+      |        "value": "max-age=315360000"
+      |      },
+      |      {
+      |        "pattern": ".*",
+      |        "value": "max-age=3600"
+      |      }
+      |    ]
+    """.stripMargin
+  )
+
+  val mimeTypes = Param[Map[String,String]]("mimeTypes",
```

Comment:
As mentioned elsewhere this is not actually consumed by the task at present so might be best to remove.

Reply:
This isn't used so the parameter can be removed.